airflow/serialization/serialized_objects.py

```diff
@@ -808,10 +808,10 @@ def serialize_task_group(cls, task_group: TaskGroup) -> Optional[Union[Dict[str,
                 else (DAT.TASK_GROUP, SerializedTaskGroup.serialize_task_group(child))
                 for label, child in task_group.children.items()
             },
-            "upstream_group_ids": cls._serialize(list(task_group.upstream_group_ids)),
-            "downstream_group_ids": cls._serialize(list(task_group.downstream_group_ids)),
-            "upstream_task_ids": cls._serialize(list(task_group.upstream_task_ids)),
-            "downstream_task_ids": cls._serialize(list(task_group.downstream_task_ids)),
+            "upstream_group_ids": cls._serialize(sorted(list(task_group.upstream_group_ids))),
```

Comment:
```suggestion
            "upstream_group_ids": cls._serialize(sorted(task_group.upstream_group_ids)),
```

Same for others below.

Reply:
fixed